### PR TITLE
feat: Added max requests parameter to jobs

### PR DESCRIPTION
**Pull request by jordanemedlock**
Added a `maxRequests` setting to both the `discover` and `smartRecommendations` jobs to limit the number of requests made during those jobs.

### CODE REVIEW

File: src/service/index.ts [hunk 1]
```diff
@@ -31,6 +32,10 @@ const processMovieResult = async (movies: MovieResult[], overseerr: OverseerrApi
 
                     await overseerr.requestMovie(movie.id, options);
                 }
+                numAdded += 1;
+                if (maxRequests !== undefined && numAdded >= maxRequests) {
+                    break;
```
Comment (psyko-gh) [inline src/service/index.ts:37]: I'd recommend adding a log statement
```suggestion
                    logger.info(`Maximum requests limit (${maxRequests}) reached. Stopping analyze here...`);
                    break;
```

File: src/service/index.ts [hunk 2]
```diff
@@ -56,6 +61,10 @@ export const discover = async () => {
         const ruleset = getRuleset(settings.ruleset);
         logger.info(`Using ruleset: ${ruleset.name}`);
 
+        if (settings.maxRequests !== undefined) {
+            logger.info(`Max requests: ${settings.maxRequests}`)
+        }
+
```
Comment (psyko-gh) [inline src/service/index.ts:67]: This log statement should be moved at the beginning of the `processMovieResult` function, so it's also logged when running the smartRecommendation job.

**Review by psyko-gh — CHANGES_REQUESTED** [review]
Thanks for the PR 😃 . Could you please consider my comments ?

File: src/service/index.ts [hunk 3]
```diff
@@ -6,7 +6,8 @@ import Ruleset, { getRuleset } from '@core/lib/ruleset';
 import { color, distinctMovies, isFulfilled, isMovie, success } from '@core/lib/utils';
 import PlexApi from '@core/api/plex';
 
-const processMovieResult = async (movies: MovieResult[], overseerr: OverseerrApi, ruleset: Ruleset, dryRun: boolean) => {
+const processMovieResult = async (movies: MovieResult[], overseerr: OverseerrApi, ruleset: Ruleset, dryRun: boolean, maxRequests: number | undefined) => {
+    var numAdded = 0;
```
Comment (psyko-gh) [inline src/service/index.ts:10]: `var`should be replaced by `let`

**Comment by jordanemedlock** [comment]
Thank you for the feedback! Updated all requests :-)

**Comment by psyko-gh** [comment]
Format check is failing. Can you run the `npm run format:fix` command please ?

**Pull request merged**